src/cuda/graph_ops.cu

```diff
@@ -1,77 +1,133 @@
 #include "graph_ops.h"
 
+#include "atomic.h"
 #include "cuda_common.h"
 #include "utils.h"
 
 namespace gs {
 namespace impl {
+template <typename IdType, typename DType>
+__global__ void GroupSum(IdType* indptr, IdType* e_ids, DType* data,
+                         DType* out_data, int64_t size) {
+  int64_t row = blockIdx.x * blockDim.y + threadIdx.y;
 
-template <typename IdType>
-torch::Tensor GetSubIndptr(torch::Tensor indptr, torch::Tensor column_ids) {
-  int64_t size = column_ids.numel();
-  auto new_indptr = torch::zeros(size + 1, indptr.options());
-  thrust::device_ptr<IdType> item_prefix(
-      static_cast<IdType*>(new_indptr.data_ptr<IdType>()));
+  while (row < size) {
+    IdType start = indptr[row];
+    IdType n_edges = indptr[row + 1] - start;
+    for (int idx = threadIdx.x; idx < n_edges; idx += blockDim.x) {
+      IdType pos = (e_ids == nullptr) ? start + idx : e_ids[start + idx];
+      AtomicAdd(&out_data[row], (data == nullptr) ? 1.0 : data[start + idx]);
```

Comment:
Would be better to use a local variable to aggregate first to reduce the number of AtomicAdd. For example,
```
DType res = 0;
for (int idx = ...) {
  res += ...
}
AtomicAdd(xxx, res);
```